src/env.rs

```diff
@@ -183,3 +184,22 @@ pub fn http_timeout() -> Result<Duration, Error> {
         )
         .map_err(Error::HttpTimeout)
 }
+
+pub fn max_paste_expiry() -> Option<u32> {
+    std::env::var(VAR_MAX_PASTE_EXPIRY)
+        .ok()
+        .and_then(|raw_max_exp| {
+            match raw_max_exp.parse::<i64>() {
+                Ok(max_exp) if max_exp == -1 => None,
+                Ok(max_exp) if max_exp >= u32::MAX as i64 => {
+                    tracing::warn!("the expiry is larger than an u32 ({} seconds, ~136 Years), assuming you mean no timeout", u32::MAX);
```

Comment:
Just fail startup (i.e. return `Result<Option<u32>>`). People ignore warnings and will complain if things do not behave as they erroneously assumed.